src/components/Calculator.js

```diff
@@ -1,54 +1,79 @@
 /* eslint-disable no-unused-vars */
 import React from 'react';
 import Buttons from './Button';
+import calculate from './logics/calculate';
 
 /* eslint-disable class-methods-use-this */
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { display: '0' };
+    this.state = {
+      display: '0',
+      currentOp: {
+        total: '0',
+        next: null,
+        operation: null,
+      },
+    };
     this.clickHandler = this.clickHandler.bind(this);
   }
 
   clickHandler = (e) => {
-    this.setState({ display: e.target.value });
+    const { currentOp } = this.state;
+    this.setState({ total: e.target.value });
+    const result = calculate(currentOp, e.target.value);
+    this.setState({ currentOp: this.updateOperation(result) });
+  };
+
+  updateOperation = ({ next, total, operation }) => {
+    const { currentOp } = this.state;
+    if (total || total === null) {
+      currentOp.total = total;
+    }
+    if (next || next === null) {
+      currentOp.next = next;
+    }
+    if (operation || operation === null) {
+      currentOp.operation = operation;
+    }
+    return currentOp;
   }
 
   render() {
     return (
       <table>
         <tbody>
           <tr>
-            <td colSpan="4" className="display">{this.state.display}</td>
+            <td colSpan="4" className="display">{this.state.currentOp.next === null ? this.state.currentOp.total : this.state.currentOp.next}</td>
           </tr>
           <tr>
-            <td><Buttons number="AC" /></td>
-            <td><Buttons number="+/-" /></td>
-            <td><Buttons number="%" /></td>
-            <td className="orange"><Buttons number="÷" /></td>
+            <td><Buttons number="AC" clickHandler={this.clickHandler} /></td>
+            <td><Buttons number="+/-" clickHandler={this.clickHandler} /></td>
+            <td><Buttons number="%" clickHandler={this.clickHandler} /></td>
+            <td className="orange"><Buttons number="÷" clickHandler={this.clickHandler} /></td>
           </tr>
           <tr>
-            <td><Buttons number="7" /></td>
-            <td><Buttons number="8" /></td>
-            <td><Buttons number="9" /></td>
-            <td className="orange"><Buttons number="*" /></td>
+            <td><Buttons number="7" clickHandler={this.clickHandler} /></td>
+            <td><Buttons number="8" clickHandler={this.clickHandler} /></td>
+            <td><Buttons number="9" clickHandler={this.clickHandler} /></td>
+            <td className="orange"><Buttons number="*" clickHandler={this.clickHandler} /></td>
```

Comment:
- [ ] The multiply operation does not work:

![image](https://user-images.githubusercontent.com/57421823/155608695-26ecc349-040a-417e-90d1-b65fc83dfcca.png)

Kindly correct this functionality so your calculator works as intended.

## Explanation and hints:

- You are using here "*" instead of "x", as stated on `operate.js`.  

Look at `operate.js`:

![image](https://user-images.githubusercontent.com/57421823/155609353-a27aac31-6e2e-4b58-a248-692f558ff642.png)

BUT, on your Calculator.js, you are referring to that operation as `*`:

![image](https://user-images.githubusercontent.com/57421823/155609706-f4e982d4-7cef-4ae9-8ff4-915f3af99946.png)


Just swap your `*` for `x` on line 59 of Calculator.js and it should work!  Try it! 💪  
I hope this is helpful to you.